src/Microsoft.Azure.SignalR.Common/IServiceEndpointGenerator.cs

```diff
@@ -1,12 +1,14 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using Microsoft.AspNetCore.Http;
+
 namespace Microsoft.Azure.SignalR
 {
     internal interface IServiceEndpointGenerator
     {
         string GetClientAudience(string hubName);
-        string GetClientEndpoint(string hubName, string originalPath);
+        string GetClientEndpoint(string hubName, string originalPath, QueryString queryString);
```

Comment:
>QueryString [](start = 70, length = 11)

how about using `string` instead of `QueryString` to get rid of the `Microsoft.AspNetCore.Http` dependency?

Reply:
It is doable. Just with `QueryString`, we can save quite some checks.